crates/bevy_core_pipeline/src/motion_blur/pipeline.rs

```diff
@@ -0,0 +1,258 @@
+                count: None,
+            },
+            // The settings uniform that will control the effect
+            BindGroupLayoutEntry {
+                binding: 4,
+                visibility: ShaderStages::FRAGMENT,
+                ty: BindingType::Buffer {
+                    ty: BufferBindingType::Uniform,
+                    has_dynamic_offset: false,
+                    min_binding_size: Some(MotionBlur::min_size()),
+                },
+                count: None,
+            },
+            // Global uniforms
+            BindGroupLayoutEntry {
+                binding: 5,
+                visibility: ShaderStages::FRAGMENT,
+                ty: BindingType::Buffer {
+                    ty: BufferBindingType::Uniform,
+                    has_dynamic_offset: false,
+                    min_binding_size: Some(GlobalsUniform::min_size()),
+                },
+                count: None,
+            },
+        ];
+
+        let sampler = render_device.create_sampler(&SamplerDescriptor::default());
+        let layout = render_device.create_bind_group_layout("motion_blur_layout", mb_layout);
+        let layout_msaa =
+            render_device.create_bind_group_layout("motion_blur_layout_msaa", mb_layout_msaa);
+
+        Self {
+            sampler,
+            layout,
+            layout_msaa,
+        }
+    }
+}
+
+#[derive(PartialEq, Eq, Hash, Clone, Copy)]
+pub struct MotionBlurPipelineKey {
+    hdr: bool,
+    samples: u32,
+}
+
+impl SpecializedRenderPipeline for MotionBlurPipeline {
+    type Key = MotionBlurPipelineKey;
+
+    fn specialize(&self, key: Self::Key) -> RenderPipelineDescriptor {
+        let layout = match key.samples {
+            1 => vec![self.layout.clone()],
+            _ => vec![self.layout_msaa.clone()],
+        };
+
+        let mut shader_defs = vec![];
+
+        match key.samples {
+            1 => (),
+            _ => shader_defs.push(ShaderDefVal::from("MULTISAMPLED")),
+        }
+
+        #[cfg(all(feature = "webgl", target_arch = "wasm32"))]
+        {
+            shader_defs.push("NO_ARRAY_TEXTURES_SUPPORT".into());
+            shader_defs.push("NO_DEPTH_TEXTURE_SUPPORT".into());
+            shader_defs.push("SIXTEEN_BYTE_ALIGNMENT".into());
+        }
+
+        RenderPipelineDescriptor {
+            label: Some("motion_blur_pipeline".into()),
+            layout,
+            vertex: fullscreen_shader_vertex_state(),
+            fragment: Some(FragmentState {
+                shader: MOTION_BLUR_SHADER_HANDLE,
+                shader_defs,
+                entry_point: "fragment".into(),
+                targets: vec![Some(ColorTargetState {
+                    format: if key.hdr {
+                        ViewTarget::TEXTURE_FORMAT_HDR
+                    } else {
+                        TextureFormat::bevy_default()
+                    },
+                    // blend: Some(BlendState::REPLACE),
+                    blend: None,
```

Comment:
Why is this None?

Reply:
Because that's what all the other similar pipelines do.

Reply:
None is faster